impl/src/main/java/com/okta/sdk/impl/util/OAuth2Utils.java

```diff
@@ -0,0 +1,163 @@
+                return responseMap.get("access_token");
+            }
+            else {
+                String error = responseMap.get("error");
+                String error_description = responseMap.get("error_description");
+                if (error != null && error.length() > 0 &&
+                    error_description != null && error_description.length() > 0) {
+                    throw new HttpException("Received HTTP " + accessTokenResponse.getHttpStatus() +
+                        " response from Authorization Server. error - " + error +
+                        ", error_description - " + error_description);
+                }
+                else {
+                    throw new HttpException("Received HTTP " + accessTokenResponse.getHttpStatus() +
+                        " response from Authorization Server");
+                }
+            }
+        }
+        else {
+            throw new HttpException("Could not retrieve access token from Authorization Server");
+        }
+    }
+
+    public static void validateOAuth2ClientConfig(ClientConfiguration clientConfiguration)
+        throws IllegalArgumentException {
+        if (clientConfiguration.getScopes() == null || clientConfiguration.getScopes().size() == 0) {
+            throw new IllegalArgumentException("Missing scopes");
+        }
+
+        if (clientConfiguration.getKeyFilePath() == null) {
+            throw new IllegalArgumentException("Missing keyFilePath");
+        }
+
+        if (!allowedSigningAlgorithms.contains(clientConfiguration.getAlgorithm())) {
+            throw new IllegalArgumentException("Unsupported algorithm");
+        }
+    }
+
+    static SignatureAlgorithm getSignatureAlgorithm(String algorithm) throws NoSuchAlgorithmException {
+        switch (algorithm) {
+            case "RS256":
+                return SignatureAlgorithm.RS256;
+
+            case "RS384":
+                return SignatureAlgorithm.RS384;
+
+            case "RS512":
+                return SignatureAlgorithm.RS512;
+
+            case "ES256":
+                return SignatureAlgorithm.ES256;
+
+            case "ES384":
+                return SignatureAlgorithm.ES384;
+
+            case "ES512":
+                return SignatureAlgorithm.ES512;
+
+            default:
+                throw new NoSuchAlgorithmException("Unsupported algorithm '" + algorithm + "'");
+        }
+    }
+
+    static PrivateKey createPrivateKeyFromPemFilePath(final String keyFileName) throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+        PemReader pemReader = new PemReader(new FileReader(keyFileName));
+        PemObject pemObject = pemReader.readPemObject();
+        byte[] pemContent = pemObject.getContent();
+        pemReader.close();
+        PKCS8EncodedKeySpec encodedKeySpec = new PKCS8EncodedKeySpec(pemContent);
+        KeyFactory keyFactory = getKeyFactoryInstance();
```

Comment:
You _might_ be able to use this keyspec as is instead of trying to extract the private key out of it.  (I've you have already tried that disregard)

Reply:
Looks like I need to generate privateKey this way. I realized that I need to update code to check for RSA or EC and then generate private key accordingly. Will update code to effect the same.

Reply:
You _should_ be able to get this directly from the PEM file, use something like this:
https://stackoverflow.com/a/60109600/6415199